src/explore-education-statistics-admin/src/pages/release/datablocks/components/chart/__tests__/ChartBuilder.test.tsx

```diff
@@ -0,0 +1,321 @@
+
+    expect(
+      screen.getByRole('button', { name: 'Chart preview' }),
+    ).toBeInTheDocument();
+
+    expect(
+      screen.getByRole('tab', { name: 'Chart configuration' }),
+    ).toBeInTheDocument();
+    expect(screen.getByRole('tab', { name: 'Data sets' })).toBeInTheDocument();
+    expect(screen.getByRole('tab', { name: 'Legend' })).toBeInTheDocument();
+    expect(
+      screen.getByRole('tab', { name: 'X Axis (major axis)' }),
+    ).toBeInTheDocument();
+    expect(
+      screen.getByRole('tab', { name: 'Y Axis (minor axis)' }),
+    ).toBeInTheDocument();
+
+    expect(
+      screen.getByRole('heading', { name: 'Chart configuration' }),
+    ).toBeInTheDocument();
+  });
+
+  describe('data sets', () => {
+    test('adding data sets', async () => {
+      render(
+        <ChartBuilderFormsContextProvider initialForms={testFormState}>
+          <ChartBuilder
+            releaseId="release-1"
+            data={testTable.results}
+            meta={testSubjectMeta}
+            tableTitle="Table title"
+            onChartSave={jest.fn(() => Promise.resolve())}
+            onChartDelete={noop}
+            onTableQueryUpdate={jest.fn(() => Promise.resolve())}
+          />
+        </ChartBuilderFormsContextProvider>,
+      );
+
+      userEvent.click(screen.getByRole('button', { name: 'Line' }));
+
+      userEvent.click(screen.getByRole('tab', { name: 'Data sets' }));
+
+      expect(
+        screen.getByRole('heading', { name: 'Data sets' }),
+      ).toBeInTheDocument();
+
+      userEvent.selectOptions(screen.getByLabelText('Characteristic'), 'male');
+      userEvent.selectOptions(
+        screen.getByLabelText('Indicator'),
+        'unauthorised-absence-sessions',
+      );
+
+      userEvent.click(screen.getByRole('button', { name: 'Add data set' }));
+
+      await waitFor(() => {
+        expect(screen.getByText('Remove all')).toBeInTheDocument();
+      });
+
+      const tableRows = screen.getAllByRole('row');
+      expect(tableRows).toHaveLength(2);
+      expect(tableRows[1]).toHaveTextContent(
+        'Number of unauthorised absence sessions (Male, All locations, All time periods)',
+      );
+
+      userEvent.selectOptions(
+        screen.getByLabelText('Characteristic'),
+        'female',
+      );
+      userEvent.selectOptions(
+        screen.getByLabelText('Indicator'),
+        'unauthorised-absence-sessions',
+      );
+
+      userEvent.click(screen.getByRole('button', { name: 'Add data set' }));
+
+      await waitFor(() => {
+        expect(screen.getByText('Reorder data sets')).toBeInTheDocument();
+      });
+
+      const updatedTableRows = screen.getAllByRole('row');
+      expect(updatedTableRows).toHaveLength(3);
+      expect(updatedTableRows[2]).toHaveTextContent(
+        'Number of unauthorised absence sessions (Female, All locations, All time periods)',
+      );
+    });
+
+    test('removing data sets', async () => {
```

Comment:
Would split out two test cases to make this a little cleaner - one for removing a single data set and one for removing all data sets.